ckanext/external_storage/react/components/FileInputComponent.js

```diff
@@ -0,0 +1,243 @@
+              >
+                {ckan.i18n._('Remove')}
+              </button>
+            </span>
+          </div>
+        </div>
+      );
+    }
+  }
+}
+
+function FileUploader({ handleFileSelected, setUploadType }) {
+  const { getRootProps, getInputProps, open } = useDropzone({
+    multiple: false,
+    noClick: true,
+    onDrop: acceptedFiles =>
+      handleFileSelected(acceptedFiles[0]),
+    onDropRejected: rejectedFiles =>
+      handleFileSelected(rejectedFiles[0].file),
+  })
+  const uploadOptions = [
+    {
+      label: ckan.i18n._('Upload a file'),
+      icon: 'fa-cloud-upload',
+      onClick: e => {
+        open(e);
+        e.preventDefault();
+      }
+    },
+    {
+      label: ckan.i18n._('Link'),
+      icon: 'fa-globe',
+      onClick: e => {
+        setUploadType('url');
+        e.preventDefault();
+      }
+    }
+  ]
+  return (
+    <div {...getRootProps({ className: 'dropzone' })}>
+      <input {...getInputProps()} />
+      <p>{ckan.i18n._('Drag a file into this box or')}</p>
+      <div className="btn-group">
+        {uploadOptions.map(option => (
+          <button
+            key={option.label}
+            className="btn btn-default"
+            onClick={option.onClick}
+          >
+            <i className={`fa ${option.icon}`}></i>
+            {option.label}
+          </button>
+        ))}
+      </div>
+    </div>
+  )
+}
+
+function ProgressBar({ uploadProgress, hiddenInputs, resetFileUploader }) {
+  const percent = Math.round(
+    (uploadProgress.loaded / uploadProgress.total) * 100
+  );
+  const loaded = percent === 100 && hiddenInputs.name;
+  return (
+    <>
+      {loaded
+        ?
+        <h3>
+          <i className="fa fa-file"></i>
+        &nbsp; {hiddenInputs.name} &nbsp;
+        <i
+            className="fa fa-close text-danger"
+            style={{ cursor: 'pointer' }}
+            title={ckan.i18n._('Remove')}
+            onClick={resetFileUploader}
+          ></i>
+        </h3>
+        :
+        <div className={`form-group controls progress ${!loaded && 'progress-striped active'}`}>
+          <div
+            className="progress-bar"
+            style={{ width: `${percent}%` }}
+          >
+            <span>{percent}%</span>
+          </div>
+        </div>
+      }
+    </>
+  )
+}
+
+const componentElement =
+  document.getElementById('FileInputComponent');
+const getAttr = key => {
+  const val = componentElement.getAttribute(`data-${key}`);
+  return ['None', ''].includes(val) ? null : val;
+};
+const requiredString = str => {
+  console.assert(str.length);
+  return str;
+}
+const
+  lfsServer = requiredString(getAttr('lfsServer')),
+  orgId = requiredString(getAttr('orgId')),
+  datasetId = requiredString(getAttr('datasetId'));
+
+let existingResourceData = {};
+if (getAttr('existingSha256')) {
+  existingResourceData = {
+    sha256: getAttr('existingSha256'),
+    name: getAttr('existingName'),
+    size: getAttr('existingSize'),
```

Comment:
Hopefully these three fields are the only things which diffirentiate between blob/link resource uploads.

Here's a diff of a **file vs link upload**
![image](https://user-images.githubusercontent.com/2634482/108754696-2854cb80-753e-11eb-8a64-4ea1ba6ea816.png)